### DISCUSSION TITLE
Implement location permission option in setting

### DISCUSSION THREAD

**Discussion by ParthTagalpallewar** [organicmaps/organicmaps#4814] [Ideas]
Hello, I am parth, second year student of Pune Institute of computer technology. I can suggest adding option to get location permission would be better then user manually going to info of application and give permission. 

Please let me know, may I implement this feature in Organic Maps. 

Thank you!

**Comment by biodranik** [comment]
Please describe in more details what exactly do you want to improve, and maybe provide some examples from other apps for clarity.

**Comment by ParthTagalpallewar** [comment]
I wanted to add option "enable location" in setting  to get permission of location from user, same as Organic-Map uses to ask on splash screen, this features is useful if user deny to give location permission earlier and later he wants to enable that location permission. 

In such cases user have to manually enable permission externally, but I am going to add option which will let user to enable location permission.

**Reply by biodranik** [reply]
Shouldn't this permission request dialog be displayed when user presses the location button, under + and -?

**Reply by ParthTagalpallewar** [reply]
I believe that it will be very inconvenient to ask user each time he/she presses + or - buttons.
I also believe that adding this feature (permission request dialogue) in menu->settings will be good to access.  

![1](https://user-images.githubusercontent.com/55437315/226175710-23950b70-4fcc-48b9-84f6-e6b65c80e726.jpeg)

![2](https://user-images.githubusercontent.com/55437315/226175721-e6a0749a-b11d-4f0a-83f5-d0563a78c2b8.jpeg)

![3](https://user-images.githubusercontent.com/55437315/226175726-d6b1fc3c-e6a4-4e89-bc5a-302f1bd11f3b.jpeg)

**Reply by biodranik** [reply]
What should be displayed/should happen in your case, if user press a crossed crosshair button that is visible on your screenshot above, below + and - buttons?

**Reply by scrunchy** [reply]
I cannot then turn off location services within the app, I can only toggle between two options: 1 - Center the map at my current location using location services, and 2 - Orient the map to north up using location services by pushing the crossed crosshair button.  

What I think could be a good option is to add an option to toggle off location services within the map - this could be a new third option that is toggled through by the button pushing, similar to the two options that are offered now .

This way if you don't want to be using battery accessing the location services you can have them off within the app.

**Reply by biodranik** [reply]
You can now do that by turning the location off in the system settings and then launching OM. If OM is closed without an enabled location, then it won't try to activate it, at least for a day.

**Reply by scrunchy** [reply]
Yes, I'd found that option and have been using that approach and found that it does work.  We used Orux maps for years and got used to having an in-app switch, so I was hoping for an in-app switch for easier access.
